src/transformers/models/vits/modeling_vits.py

```diff
@@ -1435,7 +1435,9 @@ def forward(
         if self.config.num_speakers > 1 and speaker_id is not None:
             if not 0 <= speaker_id < self.config.num_speakers:
                 raise ValueError(f"Set `speaker_id` in the range 0-{self.config.num_speakers - 1}.")
-            speaker_embeddings = self.embed_speaker(torch.tensor([speaker_id])).unsqueeze(-1)
+            if isinstance(speaker_id, int):
+                speaker_id = torch.tensor([speaker_id], device=self.device)
```

Comment:
I remember seeing somewhere that this will still copy the `speaker_id` from CPU to GPU. See #26151. Maybe `as_tensor` avoids copying?

Reply:
It seems like the correct way to make the tensor directly on device would be to do
`speaker_id = torch.full(size=(1,), fill_value=speaker_id, device=self.device)`
I've profiled the code the same way it was done in the mentioned issue

`torch.as_tensor()` and `torch.tensor()` both have `aten:to`
![image](https://github.com/huggingface/transformers/assets/32309516/6d26fa23-90c0-4e4b-86ac-a67634dd1390)

whereas `torch.full()` does not
![image](https://github.com/huggingface/transformers/assets/32309516/cd388f1a-2873-4eff-b742-3d24f48cdbe8)

Timing them in jupyter also shows that `torch.full()` is almost twice as fast
![image](https://github.com/huggingface/transformers/assets/32309516/d75ff671-03ec-4ab0-afbf-232455625e3f)

ps: thanks for updating my priors on this!